app/views/course/lectures/index.html.slim

```diff
@@ -0,0 +1,6 @@
+= page_header lectures_title do
+  = new_button([current_course, :lecture]) if can?(:create, Course::Lecture.new(course: current_course))
+
+= render partial: 'course/lectures/lecture', collection: @lectures,
```

Comment:
Remove the `to_parital_path` in Lecture model, and you only need `render @lectures`